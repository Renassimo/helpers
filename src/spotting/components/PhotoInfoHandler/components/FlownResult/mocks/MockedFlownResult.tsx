const MockedFlownResult = jest.fn(() => <div>MockedFlownResult</div>);

export default MockedFlownResult;
