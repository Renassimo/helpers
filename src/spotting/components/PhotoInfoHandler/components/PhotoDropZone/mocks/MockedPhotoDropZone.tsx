const MockedPhotoDropZone = jest.fn(() => <div>MockedPhotoDropZone</div>);

export default MockedPhotoDropZone;
