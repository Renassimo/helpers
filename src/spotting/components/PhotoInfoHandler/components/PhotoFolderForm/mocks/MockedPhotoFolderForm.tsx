const MockedPhotoFolderForm = jest.fn(() => <div> MockedPhotoFolderForm</div>);

export default MockedPhotoFolderForm;
