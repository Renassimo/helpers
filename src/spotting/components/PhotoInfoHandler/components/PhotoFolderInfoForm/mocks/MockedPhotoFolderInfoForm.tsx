const MockedPhotoFolderInfoForm = jest.fn(() => (
  <div>MockedPhotoFolderInfoForm</div>
));

export default MockedPhotoFolderInfoForm;
