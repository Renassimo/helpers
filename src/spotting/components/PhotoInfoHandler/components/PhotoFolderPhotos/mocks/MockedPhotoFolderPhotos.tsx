const MockedPhotoFolderPhotos = jest.fn(() => (
  <div>MockedPhotoFolderPhotos</div>
));

export default MockedPhotoFolderPhotos;
