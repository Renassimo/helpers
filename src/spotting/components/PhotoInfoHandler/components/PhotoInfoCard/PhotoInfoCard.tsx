import Box from '@mui/material/Box';
import CardMedia from '@mui/material/CardMedia';
import IconButton from '@mui/material/IconButton';
import ContentCopyIcon from '@mui/icons-material/ContentCopy';

import ZoomInIcon from '@mui/icons-material/ZoomIn';
import RemoveCircleOutlineIcon from '@mui/icons-material/RemoveCircleOutline';

import SelectableCard from '@/spotting/components/SelectableCard';

const PhotoInfoCard = ({
  selected = false,
  name,
  preview,
  id,
  onPhotoSelected,
  onOpenZoom,
  onClick,
  onRemoveFromFolder,
  onDuplicate,
}: {
  selected?: boolean;
  name: string;
  preview: string;
  id: string;
  onPhotoSelected?: (id: string) => void;
  onOpenZoom?: (id: string) => void;
  onClick?: (id?: string) => void;
  onRemoveFromFolder?: (id: string) => void;
  onDuplicate?: (id: string) => void;
}) => {
  const toggleSelect = () => onPhotoSelected?.(id);
  const handleClick = (onClick && (() => onClick(id))) || toggleSelect;

  return (
    <SelectableCard selectable selected={selected} toggleSelect={toggleSelect}>
      <Box position="relative">
        <CardMedia
          image={preview ?? '/images/plane.png'}
          title={name}
          sx={{
            height: 140,
            maxWidth: '100%',
            cursor: 'pointer',
          }}
          onClick={handleClick}
        />
        <Box
          position="absolute"
          bottom={0}
          right={0}
          display="flex"
          flexDirection="column"
        >
          {onRemoveFromFolder && (
            <IconButton
              aria-label="remove"
              onClick={() => onRemoveFromFolder(id)}
            >
              <RemoveCircleOutlineIcon />
            </IconButton>
          )}
          {onDuplicate && (
            <IconButton aria-label="duplicate" onClick={() => onDuplicate(id)}>
              <ContentCopyIcon />
            </IconButton>
          )}
          {onOpenZoom && (
            <IconButton aria-label="zoom" onClick={() => onOpenZoom(id)}>
              <ZoomInIcon />
            </IconButton>
          )}
        </Box>
      </Box>
    </SelectableCard>
  );
};

export default PhotoInfoCard;
