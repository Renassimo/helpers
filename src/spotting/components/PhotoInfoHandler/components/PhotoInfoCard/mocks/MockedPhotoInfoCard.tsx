const MockedPhotoInfoCard = jest.fn(
  ({
    selected,
    name,
    preview,
    path,
    onPhotoSelected,
    onOpenZoom,
    onClick,
    onRemoveFromFolder,
  }) => (
    <div>
      {selected && 'selected'} - {name} - {preview} - {path} -{' '}
      {!!onPhotoSelected && 'onPhotoSelected'} - {!!onOpenZoom && 'onOpenZoom'}{' '}
      - {!!onClick && 'onClick'} -{' '}
      {!!onRemoveFromFolder && 'onRemoveFromFolder'}
    </div>
  )
);

export default MockedPhotoInfoCard;
