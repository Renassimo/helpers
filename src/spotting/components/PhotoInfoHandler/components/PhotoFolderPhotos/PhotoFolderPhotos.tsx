import { memo, useCallback } from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';

import { PhotoActionType } from '@/spotting/types';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

import PhotoInfoCard from '@/spotting/components/PhotoInfoHandler/components/PhotoInfoCard';

const MemorizedCard = memo(PhotoInfoCard);

const PhotoFolderPhotos = () => {
  const { showingFolder, dispatch } = usePhotoInfoContext();

  const onOpenZoom = useCallback(
    (path: string) =>
      dispatch({ type: PhotoActionType.SET_ZOOMED_PHOTO, payload: path }),
    []
  );
  const onRemoveFromFolder = useCallback(
    (path: string) =>
      dispatch({ type: PhotoActionType.REMOVE_FROM_FOLDER, payload: path }),
    []
  );
  const onRemoveFolder = useCallback(
    (id: string) =>
      dispatch({ type: PhotoActionType.REMOVE_FOLDER, payload: id }),
    []
  );
  const onDuplicatePhoto = useCallback(
    (id: string) =>
      dispatch({ type: PhotoActionType.DUPLICATE_PHOTO, payload: id }),
    []
  );

  return (
    <Box width="100%">
      {showingFolder && (
        <Box>
          <>
            <Button onClick={() => onRemoveFolder?.(showingFolder?.id)}>
              Remove folder
            </Button>
          </>
        </Box>
      )}
      <Grid container spacing={2} py={2}>
        {showingFolder &&
          Object.values(showingFolder.photos).map((photo) => (
            <Grid
              maxWidth={300}
              key={photo.path}
              item
              xs={12}
              sm={6}
              md={4}
              lg={3}
            >
              <MemorizedCard
                selected={photo.selected}
                name={photo.name}
                preview={photo.preview}
                id={photo.path}
                onOpenZoom={onOpenZoom}
                onRemoveFromFolder={onRemoveFromFolder}
                onDuplicate={onDuplicatePhoto}
              />
            </Grid>
          ))}
      </Grid>
    </Box>
  );
};

export default PhotoFolderPhotos;
