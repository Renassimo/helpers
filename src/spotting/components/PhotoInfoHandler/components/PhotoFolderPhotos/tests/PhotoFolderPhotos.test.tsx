import renderWithTheme from '@/common/tests/helpers/renderWithTheme';
import userEvent from '@testing-library/user-event';

import usePhotoInfoContext from '@/spotting/contexts/hooks/usePhotoInfoContext';

import { PhotoActionType } from '@/spotting/types';

import MockedPhotoInfoCard from '@/spotting/components/PhotoInfoHandler/components/PhotoInfoCard/mocks';
import { mockedPhotoFolder } from '@/spotting/types/mocks';

import PhotoInfoCard from '@/spotting/components/PhotoInfoHandler/components/PhotoInfoCard';

import PhotoFolderPhotos from '../PhotoFolderPhotos';

jest.mock('@/spotting/components/PhotoInfoHandler/components/PhotoInfoCard');
jest.mock('@/spotting/contexts/hooks/usePhotoInfoContext');

describe('PhotoFolderPhotos', () => {
  const dispatch = jest.fn();

  beforeEach(() => {
    (PhotoInfoCard as unknown as jest.Mock).mockImplementation(
      MockedPhotoInfoCard
    );
    (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
      jest.fn(() => ({ dispatch, showingFolder: null }))
    );
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<PhotoFolderPhotos />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when photoFolder chosen', () => {
    beforeEach(() => {
      (usePhotoInfoContext as unknown as jest.Mock).mockImplementation(
        jest.fn(() => ({ dispatch, showingFolder: mockedPhotoFolder }))
      );
    });

    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(<PhotoFolderPhotos />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });

    describe('when Remove folder clicked', () => {
      test('calls dispatch', async () => {
        // Arange
        const { getByText } = renderWithTheme(<PhotoFolderPhotos />);
        // Act
        await userEvent.click(getByText('Remove folder'));
        // Assert
        expect(dispatch).toBeCalledWith({
          type: PhotoActionType.REMOVE_FOLDER,
          payload: mockedPhotoFolder.id,
        });
      });
    });
  });
});
