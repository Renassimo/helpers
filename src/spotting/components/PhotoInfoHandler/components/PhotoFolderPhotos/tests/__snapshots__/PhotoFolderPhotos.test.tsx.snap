// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoFolderPhotos renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1pq723i-MuiGrid-root"
      />
    </div>
  </div>
</body>
`;

exports[`PhotoFolderPhotos when photoFolder chosen renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Remove folder
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1pq723i-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-t2nu3-MuiGrid-root"
        >
          <div>
            selected
             - 
            name1
             - 
            preview1
             - 
             -
             
             - 
            onOpenZoom
             
            - 
            onClick
             -
             
            onRemoveFromFolder
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-t2nu3-MuiGrid-root"
        >
          <div>
            selected
             - 
            name2
             - 
            preview2
             - 
             -
             
             - 
            onOpenZoom
             
            - 
            onClick
             -
             
            onRemoveFromFolder
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
